r/R/duckdb.R

```diff
@@ -111,3 +111,18 @@ duckdb_disconnector <- function(con, tbl_name) {
   })
   environment()
 }
+
+to_arrow <- function(.data) {
+  res <- DBI::dbSendQuery(dbplyr::remote_con(.data), dbplyr::remote_query(.data), arrow = TRUE)
+
+  # transform the source ExecNode we get below into something dataset-like that we can
+  # make arrow_dplyr_querys/datasets from
+  # Should be able to:
+  # * get the record batch reader from duckdb
+  # * produce the SourceNode
+  # * build an ExecPlan with that in place of the ScanNode you would have gotten from ExecNode_Scan
+  plan <- ExecPlan$create()
```

Comment:
I don't think this will work because you're making this node with an ExecPlan here, but I think you'll be creating a different ExecPlan in `collect()`. 

Why not return the RecordBatchReader from duckdb::duckdb_fetch_record_batch(res)? Then do the wrapping of that inside `plan$Build` (or even `plan$Scan`)

Reply:
Oh that's an interesting idea. Let me explore that. AFAIU we can't _simply_ return the RecordBatchReader, since we want to be able to continue building an `arrow_dplyr_query` here (unless we make all the dplyr methods available for RecordBatchReaders too)

Reply:
Right. So to restate: allow an arrow_dplyr_query to hold a RecordBatchReader in .data (similar to how I extended it recently to allow holding an arrow_dplyr_query itself, in collapse). So you'd return an arrow_dplyr_query here, containing a RecordBatchReader.